app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt

```diff
@@ -183,28 +182,54 @@ fun ActivityDetailsScreen(
 
               // Enroll button
               if (activity?.status == ActivityStatus.ACTIVE && profile != null) {
-                Button(
-                    onClick = {
-                      if (((placesTaken ?: 0) >= 0) && ((placesTaken ?: 0) < (maxPlaces ?: 0))) {
-                        val theActivity =
-                            activity.copy(placesLeft = min((placesTaken ?: 0) + 1, maxPlaces ?: 0))
-                        listActivityViewModel.updateActivity(theActivity)
-                        profileViewModel.addActivity(profile.id, theActivity.uid)
-                        Toast.makeText(context, "Enroll Successful", Toast.LENGTH_SHORT).show()
-                        navigationActions.navigateTo(Screen.OVERVIEW)
-                      } else {
-                        Toast.makeText(
-                                context,
-                                "Enroll failed, limit of places reached",
-                                Toast.LENGTH_SHORT)
-                            .show()
+                if (activity.creator != profile.id) {
+                  Button(
+                      onClick = {
+                        if (((placesTaken ?: 0) >= 0) && ((placesTaken ?: 0) < (maxPlaces ?: 0))) {
+                          val theActivity =
+                              activity.copy(
+                                  placesLeft = min((placesTaken ?: 0) + 1, maxPlaces ?: 0))
+                          listActivityViewModel.updateActivity(theActivity)
+                          profileViewModel.addActivity(profile.id, theActivity.uid)
+                          Toast.makeText(context, "Enroll Successful", Toast.LENGTH_SHORT).show()
+                          navigationActions.navigateTo(Screen.OVERVIEW)
+                        } else {
+                          Toast.makeText(
+                                  context,
+                                  "Enroll failed, limit of places reached",
+                                  Toast.LENGTH_SHORT)
+                              .show()
+                        }
+                      },
+                      modifier =
+                          Modifier.fillMaxWidth()
+                              .padding(horizontal = 24.dp)
+                              .testTag("enrollButton")) {
+                        Text(text = "Enroll")
+                      }
+                } else {
```

Comment:
good that the creator of the activity can navigate to the editActivityScreen to edit his activity.